### PR TITLE
Fix android shutter sound

### PR DESCRIPTION
**Pull request by cristianoccazinsp**
# Summary

Fixes for Camera1 and Camera2 shutter sound. (https://github.com/react-native-community/react-native-camera/issues/2747). This includes making Camera1 and Camera2 behave consistently (when possible) and fixing MediaActionSound being instantiated each time and failing to play sound from time to time.

Shutter sound should now be consistent with the `playSoundOnCapture` prop. Additionally, it should happen on successful capture as opposed to camera capture start (consistent with iOS and any camera app)

Lastly, added missing props to the docs (https://github.com/react-native-community/react-native-camera/issues/2727), fixed issue related to double camera start (https://github.com/react-native-community/react-native-camera/issues/2521), and improved Camera2 to Camera1 fallback logic so Camera2 is not instantiated unnecessarily.

@noambonnie if you could give this PR a try. You shall now be able to use Camera1 and also `playSoundOnCapture=false` as long as your country allows for shutter sound to be disabled.

## Test Plan
Tested on Pixel 2 (Android 10) and Motorola G5 (Android 8).  Tested both Camera1 and Camera2 implementations. 

### What's required for testing (prerequisites)?
Real device

### What are the steps to reproduce (after prerequisites)?
Test `playSoundOnCapture` true and false

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅     |

## Checklist

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [X] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typed files (TS and Flow)
- [X] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: android/src/main/java/com/google/android/cameraview/CameraView.java [hunk 1]
```diff
@@ -311,7 +313,9 @@ public void setUsingCamera2Api(boolean useCamera2) {
             }
             mImpl = new Camera1(mCallbacks, mImpl.mPreview, mBgHandler);
         }
-        start();
+        if(wasOpened){
```
Comment (cristianoccazinsp) [inline android/src/main/java/com/google/android/cameraview/CameraView.java]: The following change is causing issues on devices that fallback to Camera1 even when Camera2 is used.... Implementing further fixes for those.

Comment (cristianoccazinsp) [reply android/src/main/java/com/google/android/cameraview/CameraView.java]: Changes done.

File: android/src/main/java/com/google/android/cameraview/Camera2.java [hunk 2]
```diff
@@ -727,19 +770,16 @@ private boolean chooseCameraIdByFacing() {
                 int internalFacing = INTERNAL_FACINGS.get(mFacing);
                 final String[] ids = mCameraManager.getCameraIdList();
                 if (ids.length == 0) { // No camera
-                    throw new RuntimeException("No camera available.");
+                    Log.e(TAG, "No cameras available.");
```
Comment (MateusAndrade) [inline android/src/main/java/com/google/android/cameraview/Camera2.java]: What do you think about returning something to the JS? Some people don't know how to see the `Logcat` provided  by android. Removing the `throw`(that probably show which error triggered that) can lead some confusion.

**Comment by cristianoccazinsp** [comment]
I believe the mount error callback will fire when that error happens. The
change was to prevent full app crashes when used on devices with no cameras
(making it behave like Camera1).

I agree about sending these errors to the JS side, but neither camera1 nor
2 have a way to pass down this error message through the mount error
callback. There would be a bunch of changes involved to do so (on both
camera implementations).  We could make another PR for that change, but
keep in mind both camera implementations needs to be changed.

**Comment by MateusAndrade** [comment]
>  We could make another PR for that change, but keep in mind both camera implementations needs to be changed. 

Totally agree with that. Also, we will need to change the same behaviour on `ios`. Btw, this looks good for me. What do you think @sibelius , @fabriziobertoglio1987 ?

**Comment by cristianoccazinsp** [comment]
It would be great to have @noambonnie test it on the Pixel 3 which seemed to be a problematic device.

**Comment by noambonnie** [comment]
I tested this on the following devices/versions:
Galaxy S10/Android 10
Galaxy S8/Android 9
Pixel 3a/Android 10

- I can confirm the sound flag works as expected. 
- I can confirm this resolves the issue discussed in https://github.com/react-native-community/react-native-camera/issues/2521

I would love to see this merged. 

Thank you for your help!

* Unfortunately due to COVID19 I don't currently have access to other devices. I will try to go to the office sometime this week to test, unless someone else has access to older or different devices/versions.

**Comment by fabriziobertoglio1987** [comment]
@MateusAndrade thanks sorry... I am not feeling 100% and I was trying to complete my work on another pull request... I see this is already merged so I will work on the other pull requests. Thanks a lot

**Comment by MateusAndrade** [comment]
> @MateusAndrade thanks sorry... I am not feeling 100% and I was trying to complete my work on pull [2751](https://github.com/react-native-community/react-native-camera/pull/2751)... I see this is already merged so I will review the other pull requests. Thanks a lot

Sorry for that, i was thinking that we already got your review 😢

**Pull request merged**